lib/net/ldap/connection.rb

```diff
@@ -31,10 +36,15 @@ def open_connection(server)
     hosts = server[:hosts]
     encryption = server[:encryption]
 
+    socket_opts = {
+      connect_timeout: server[:connect_timeout] || DefaultConnectTimeout,
+      timeout: server[:timeout] || DefaultTimeout
```

Comment:
According to [the doc](http://docs.ruby-lang.org/en/2.2.0/Socket.html#method-c-tcp), `Socket.tcp` deals with only `connect_timeout`. Does `timeout` make sense?